File: vanakel_group_management_platform_app/lib/features/maintenance/domain/entities/maintenance_plan.dart
class MaintenancePlan {
  final String id;
  final String buildingId;
  final String title;
  final String? titleEn;
  final String? titleFr;
  final String? titleNl;
  final String description;
  final String? descriptionEn;
  final String? descriptionFr;
  final String? descriptionNl;
  final String frequency; // YEARLY, MONTHLY, QUARTERLY
  final int interval;
  final DateTime startDate;
  final DateTime endDate;
  final String status; // ACTIVE, CANCELLED
  final String? syndicId;
  final DateTime createdAt;

  const MaintenancePlan({
    required this.id,
    required this.buildingId,
    required this.title,
    this.titleEn,
    this.titleFr,
    this.titleNl,
    required this.description,
    this.descriptionEn,
    this.descriptionFr,
    this.descriptionNl,
    required this.frequency,
    required this.interval,
    required this.startDate,
    required this.endDate,
    required this.status,
    this.syndicId,
    required this.createdAt,
  });
}

File: app/Models/MaintenancePlan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class MaintenancePlan extends Model
{
    protected $fillable = [
        'building_id',
        'syndic_id',
        'title',
        'title_en',
        'title_fr',
        'title_nl',
        'description',
        'description_en',
        'description_fr',
        'description_nl',
        'frequency',
        'interval',
        'start_date',
        'end_date',
        'status',
    ];

    protected $casts = [
        'start_date' => 'datetime',
        'end_date' => 'datetime',
    ];

    public function building()
    {
        return $this->belongsTo(Building::class);
    }

    public function syndic()
    {
        return $this->belongsTo(Syndic::class);
    }

    public function interventions()
    {
        return $this->hasMany(Intervention::class);
    }
}
